parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -267,22 +267,40 @@ retrycmd_nslookup() {
     current_time=$(date +%s)
     echo "Executed nslookup -timeout=$timeout -retry=0 $record for $((current_time - start_time)) seconds";
 }
-retrycmd_get_tarball() {
-    tar_retries=$1; wait_sleep=$2; tarball=$3; url=$4
-    echo "${tar_retries} retries"
-    for i in $(seq 1 $tar_retries); do
-        [ -f "$tarball" ] && tar -tzf "$tarball" && break || \
-        if [ "$i" -eq "$tar_retries" ]; then
+
+_retry_file_curl_internal() {
+    retries=$1; waitSleep=$2; timeout=$3; filePath=$4; url=$5; checksToRun=( "${@:6}" )
+    echo "${retries} file curl retries"
+    for i in $(seq 1 $retries); do 
+        # Use eval to execute the checksToRun string as a command
+        ( eval "$checksToRun" ) && break || if [ "$i" -eq "$retries" ]; then
```

Comment:
im not sure if this chaining will work exactly how you want it, I'd probably wrap the `( eval "$checksToRun" ) && break` in another pair of parens

Reply:
I think it should work fine, see for example 

```
Describe 'retry_file_curl_internal'
    It "returns 1 if checksToRun fail and retries are exhausted"
```